source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,257 @@
+.. _ref-cross-compilation:
+
+Cross-compilation
+=================
+
+Cross-compilation is the act of **compiling code** on the **build platform**
+to the **host platform**, where the compiled **executable runs**. [#]_
+
+It's needed when the host platform has limited resources (such as CPU)
+or when it's not easily accessible for development.
+
+Nix community has world-class support for cross-compilation,
+after years of hard work from our community.
+
+.. [#] Terminology for cross-compilation platforms differs between build systems,
+       Nix community has chosen to follow 
+       `autoconf terminology <https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html>`_.
+
+.. note:: macOS/Darwin is a special case, as not the whole OS is Open Source. 
+          It's only possible to cross-compile between ``aarch64-darwin`` and ``x86_64-darwin``.
+
+
+What's a target platform?
+-------------------------
+
+There's actually a third platform named target.
+
+It matters in cases where you'd like to distribute a compiler binary, 
+as you'd then like to build a compiler on the build platform, compile code on the
+target plaform and run the final executable on the host platform.
```

Comment:
This is wrong: You would run the compiler on the *host* platform and it would generate code for the *target* platform.

This is also why targetPlatform usually doesn't matter: host platform means the same for everything and for everything but build tools and compilers targetPlatform makes zero difference .